Add --repeating-pattern-payload option to the client side

xphil opened the pull request:
This option simulates payload in iperf2, which is just repetitive pattern, as
opposed to iperf3 where payload is random.
It might help in testing and reveal problems in networking gear with hardware
compression (including WiFi access points), where iperf2 and iperf3 perform
differently, just based on payload entropy.

_PLEASE NOTE the following text from the iperf3 license.  Submitting a
pull request to the iperf3 repository constitutes "[making]
Enhancements available...publicly":_

```
You are under no obligation whatsoever to provide any bug fixes, patches, or
upgrades to the features, functionality or performance of the source code
("Enhancements") to anyone; however, if you choose to make your Enhancements
available either publicly, or directly to Lawrence Berkeley National
Laboratory, without imposing a separate written license agreement for such
Enhancements, then you hereby grant the following license: a non-exclusive,
royalty-free perpetual license to install, use, modify, prepare derivative
works, incorporate into other computer software, distribute, and sublicense
such enhancements or derivative works thereof, in binary and source code form.
```

_The complete iperf3 license is available in the `LICENSE` file in the
top directory of the iperf3 source tree._

* Version of iperf3 (or development branch, such as `master` or
  `3.1-STABLE`) to which this pull request applies:

* Issues fixed (if any):

* Brief description of code changes (suitable for use as a commit message):

xphil commented:
Hi. Thanks for the review! Yes, the pattern is as you described, ASCII from '0' to '9', just like in iperf2. Sure, I'll add feature description to the documentation. About lots of whitespace in iperf_locale.c - yes, it is for long(er) option I've added. I agree, looks kind of ugly, but I haven't figured out how to make the option shorter. Will --repeat-pattern-payload, instead of --repeating-pattern-payload be OK?

bmah888 commented:
`--repeating-payload`?

Oh yeah.  Don't worry about editing the stuff under `docs/`, in particular the manual page.  I'll re-render `src/iperf3.1` and update the Web documentation myself.

xphil commented:
Thanks for helping out with documentation! I've changed --repeating-pattern-payload to --repeating-payload, added its description to the man page and undone all that whitespace. Don't hesitate to tell me if you need any other changes.

bmah888 commented:
Thanks for the changes @xphil!  So far it looks good although admittedly I haven't actually run this code yet.  I'm planning to merge this before the next release (unscheduled so far).

xphil commented:
Thanks for accepting my PR!
P.S. Just wanted to mention compilation and testing on my side: I've compiled it on Ubuntu  16.04.4 LTS with gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9), and checked packet contents with Wireshark. Haven't seen any extra compilation warnings and --repeating-payload did change packet payload as intended.

bmah888 commented:
Merged, thanks for the pull request!
